modules/transport-netty4/src/main/java/org/elasticsearch/transport/netty4/Netty4Transport.java

```diff
@@ -311,6 +307,11 @@ protected void stopInternal() {
 
         @Override
         protected void initChannel(Channel ch) throws Exception {
+            addClosedExceptionLogger(ch);
+            assert ch instanceof Netty4NioSocketChannel;
+            if (ch instanceof Netty4NioSocketChannel) {
+                NetUtils.setSaneDefaultKeepAliveOptions(((Netty4NioSocketChannel) ch).javaChannel());
```

Comment:
Is the `if` block so that this does not break in production is assertions are disabled and for some reason there is a scenario where the channel is a different type? Feels like we should just be able to cast after the assertion since this is all Elasticsearch code.

Reply:
You're right, I was overly defensive here.